compiler/luci/pass/src/RemoveRedundantTranspose.cpp

```diff
@@ -74,8 +72,14 @@ bool remove_consecutive_transpose_function(luci::CircleNode *node)
       new_const_node->at<loco::DataType::S32>(i) =
           pred_perm->at<loco::DataType::S32>(main_perm->at<loco::DataType::S32>(i));
     }
-    pred_node->perm(new_const_node);
-    replace(node).with(pred_node);
+
+    // Create New Transpose Node
+    auto new_transpose_node = g->nodes()->create<luci::CircleTranspose>();
+    new_transpose_node->dtype(target_node->dtype());
+    new_transpose_node->a(main_node);
+    new_transpose_node->perm(new_const_node);
+
+    replace(node).with(new_transpose_node);
```

Comment:
@struss Could you explain about this change more detail?
As below comment,
```cxx
*  AFTER
 *      <Optional Case>
 *
 *          |                 |                   |
 *    [CircleNode]      [CircleConst]             |
 *     (main_node)     (new_const_node)           |
 *           \               /           or  [CircleNode]
 *           [CircleTranspose]                (main_node)
 *              (pred_node)                       |
 *                   |                            |
 *
```
This seems to change the way it replace the `pred_node` with `target_node` from just replacement to creation and replacement. Is there any reason of this change?

Reply:
> @struss Could you explain about this change more detail?
> As below comment,
> 
> ```c++
> *  AFTER
>  *      <Optional Case>
>  *
>  *          |                 |                   |
>  *    [CircleNode]      [CircleConst]             |
>  *     (main_node)     (new_const_node)           |
>  *           \               /           or  [CircleNode]
>  *           [CircleTranspose]                (main_node)
>  *              (pred_node)                       |
>  *                   |                            |
>  *
> ```

Oh, I think I need to change this.

> This seems to change the way it replace the `pred_node` with `target_node` from just replacement to creation and replacement.

that's right.

>Is there any reason of this change?

To handle case that `pred_node` have multiple succs.
previous implementation only activate when `pred_node` has one succ and also check in only one branch.
but, if `pred_node` have multiple succs, it also could be shorten. so, to handle this case, I change logic.
I will update all comment that explain this logic.